slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbSlashingProtection.java

```diff
@@ -100,12 +106,40 @@ public void export(final OutputStream output) {
     }
   }
 
+  @Override
+  public void registerGenesisValidatorsRoot(final Bytes genesisValidatorsRoot) {
+    jdbi.useTransaction(
+        SERIALIZABLE,
+        h -> {
+          final Optional<Bytes> dbGvr = metadataDao.findGenesisValidatorsRoot(h);
+          final Optional<Boolean> isValidGvr = dbGvr.map(gvr -> gvr.equals(genesisValidatorsRoot));
+          if (!isValidGvr.orElse(true)) {
+            throw new IllegalStateException(
+                String.format(
+                    "Genesis validators root %s does not match slashing protection db validators root",
+                    genesisValidatorsRoot));
+          } else if (dbGvr.isEmpty()) {
+            metadataDao.insertGenesisValidatorsRoot(h, genesisValidatorsRoot);
+          }
+        });
+  }
+
+  private boolean isValidGenesisValidatorsRoot(final Bytes genesisValidatorsRoot) {
+    return jdbi.inTransaction(
+        SERIALIZABLE,
+        h -> {
+          final Optional<Bytes> dbGvr = metadataDao.findGenesisValidatorsRoot(h);
+          return dbGvr.map(gvr -> gvr.equals(genesisValidatorsRoot)).orElse(false);
+        });
+  }
+
   @Override
   public boolean maySignAttestation(
       final Bytes publicKey,
       final Bytes signingRoot,
       final UInt64 sourceEpoch,
-      final UInt64 targetEpoch) {
+      final UInt64 targetEpoch,
+      final Bytes genesisValidatorsRoot) {
```

Comment:
is GVR bytes, or bytes32? (I'm happy with Bytes, just want to confirm, as tests make it bytes32)

Reply:
done